code/modules/mining/ores_coins.dm

```diff
@@ -60,6 +61,25 @@
 	qdel(src)
 	return
 
+/obj/item/weapon/ore/glass/throw_impact(atom/hit_atom)
+	if(..() || !ishuman(hit_atom))
+		return
+	var/mob/living/carbon/human/C = hit_atom
+	if(C.head && C.head.flags_cover & HEADCOVERSEYES)
+		visible_message("<span class='danger'>[C]'s headgear blocks the sand!</span>")
+		return
+	if(C.wear_mask && C.wear_mask.flags_cover & MASKCOVERSEYES)
+		visible_message("<span class='danger'>[C]'s mask blocks the sand!</span>")
+		return
+	if(C.glasses && C.glasses.flags_cover & GLASSESCOVERSEYES)
+		visible_message("<span class='danger'>[C]'s glasses block the sand!</span>")
+		return
+	C.adjust_blurriness(6)
+	C.adjustStaminaLoss(15)//the pain from your eyes burning does stamina damage
+	C.confused += 5

```

Comment:
Confusion doesn't totally fuck your movement at low levels anymore so this isn't too bad